hibernate-core/src/test/java/org/hibernate/orm/test/mapping/inheritance/discriminator/MultiInheritanceDiscriminatorTest.java

```diff
@@ -0,0 +1,225 @@
+		cAcc1.setOwner( owner );
+		owner.getCreditAccounts().add( cAcc1 );
+
+		CreditAccount cAcc2 = new CreditAccount();
+		cAcc2.setAmount( new BigDecimal( "321.43" ).setScale( 19, RoundingMode.DOWN ) );
+		cAcc2.setOwner( owner );
+		owner.getCreditAccounts().add( cAcc2 );
+
+		DebitAccount dAcc = new DebitAccount();
+		dAcc.setAmount( new BigDecimal( "654.99" ).setScale( 19, RoundingMode.DOWN ) );
+		dAcc.setOwner( owner );
+		owner.getDebitAccounts().add( dAcc );
+
+		scope.inTransaction(
+				session ->
+						session.persist( owner )
+		);
+
+		Long ownerId = owner.getId();
+
+		scope.inTransaction(
+				session -> {
+					AccountOwner _owner = session.get( AccountOwner.class, ownerId );
+					assertEquals( 2, _owner.getCreditAccounts().size() );
+					assertEquals( "CreditAccount", _owner.getCreditAccounts().iterator().next().getClass().getSimpleName() );
+
+					assertEquals( 1, _owner.getDebitAccounts().size() );
+					assertEquals( "DebitAccount", _owner.getDebitAccounts().iterator().next().getClass().getSimpleName() );
+				}
+		);
+	}
+
+	@DomainModel(
+			annotatedClasses = {
+					AbstractAcc.class, CreditAcc.class, DebitAcc.class, AccOwner.class
+			}
+	)
+	@SessionFactory
+	@Test
+	public void testAbstractSuperClassMappingAnn(SessionFactoryScope scope) {
+		AccOwner owner = new AccOwner();
+		owner.setName( "Some account" );
+
+		CreditAcc cAcc1 = new CreditAcc();
+		cAcc1.setAmount( new BigDecimal( "123.34" ).setScale( 19, RoundingMode.DOWN ) );
+		cAcc1.setOwner( owner );
+		owner.getCreditAccs().add( cAcc1 );
+
+		CreditAcc cAcc2 = new CreditAcc();
+		cAcc2.setAmount( new BigDecimal( "321.43" ).setScale( 19, RoundingMode.DOWN ) );
+		cAcc2.setOwner( owner );
+		owner.getCreditAccs().add( cAcc2 );
+
+		DebitAcc dAcc = new DebitAcc();
+		dAcc.setAmount( new BigDecimal( "654.99" ).setScale( 19, RoundingMode.DOWN ) );
+		dAcc.setOwner( owner );
+		owner.getDebitAccs().add( dAcc );
+
+		scope.inTransaction(
+				session ->
+						session.persist( owner )
+		);
+
+		Long ownerId = owner.getId();
+
+		scope.inTransaction(
+				session -> {
+					AccOwner _owner = session.get( AccOwner.class, ownerId );
+					assertEquals( 2, _owner.getCreditAccs().size() );
+					assertEquals( "CreditAccount", _owner.getCreditAccs().iterator().next().getClass().getSimpleName() );
+
+					assertEquals( 1, _owner.getDebitAccs().size() );
+					assertEquals( "DebitAccount", _owner.getDebitAccs().iterator().next().getClass().getSimpleName() );
+				}
+		);
+	}
+
+	@Entity
+	@Table(name = "AccOwner")
+	public static class AccOwner {
+		@Id
+		@GeneratedValue
+		private Long id;
+
+		private String name;
+
+		@OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, mappedBy = "owner")
+		private Set<AbstractAcc> creditAccs = new HashSet<>();
+
+		@OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, mappedBy = "owner")
+		private Set<AbstractAcc> debitAccs = new HashSet<>();
```

Comment:
Actually, this should use the following mapping:

```suggestion
		@OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, mappedBy = "owner")
		private Set<CreditAcc> creditAccs = new HashSet<>();

		@OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, mappedBy = "owner")
		private Set<DebitAcc> debitAccs = new HashSet<>();
```

Reply:
I resolved to adding `targetEntity = CreditAcc.class`on the mapping, which is more inline with the hbm mapping, and it boils down to the same thing, right? It allwos me to keep the parent reference as with the hbm test.

Reply:
Yeah, it's essentially the same thing.